### ISSUE TITLE
Tiled Gallery: `srcset` attributes are missing for block created by transforming image blocks

### ISSUE THREAD

**Issue by brbrr** [Automattic/jetpack#12272]
#### Steps to reproduce the issue

1. Start a new post
2. Add a few image blocks
3. Convert them to Tiled Gallery by selecting them (with shift-click) 
4. Publish a post
5. Check image html tags on the frontend. `srcset` attributes are missing

#### What I expected
`srcset` attributes should be present for all images in Tiled Gallery block

#### What happened instead
`srcset` attributes are missing. 

Internal ref: p8oabR-lP-p2#comment-2891

**Comment by simison** [comment]
Initially when pressing "transform", I get tiled gallery without layout width/height applied to images:



When I just click somewhere in the editor, layout kicks in and adds those attributes to images and they take the correct shape:

If I saved between those states, the outcome is slightly different HTML stored in content.

Right after transform:

```html
<!-- wp:jetpack/tiled-gallery {"ids":[159,157]} -->
<div class="wp-block-jetpack-tiled-gallery aligncenter is-style-rectangular"><div class="tiled-gallery__gallery"><div class="tiled-gallery__row"><div class="tiled-gallery__col"><figure class="tiled-gallery__item"><img alt="" data-id="159" data-url="http://35a97fbe.eu.ngrok.io/wp-content/uploads/2019/05/pexels-photo-892548-1024x682.jpeg" src="https://i2.wp.com/35a97fbe.eu.ngrok.io/wp-content/uploads/2019/05/pexels-photo-892548-1024x682.jpeg"/></figure></div><div class="tiled-gallery__col"><figure class="tiled-gallery__item"><img alt="" data-id="157" data-url="http://35a97fbe.eu.ngrok.io/wp-content/uploads/2019/05/loro-park-orca-fish-1024x686.jpg" src="https://i0.wp.com/35a97fbe.eu.ngrok.io/wp-content/uploads/2019/05/loro-park-orca-fish-1024x686.jpg"/></figure></div></div></div></div>
<!-- /wp:jetpack/tiled-gallery -->
```

Right after layout rendering works:

```
<!-- wp:jetpack/tiled-gallery {"ids":[159,157]} -->
<div class="wp-block-jetpack-tiled-gallery aligncenter is-style-rectangular"><div class="tiled-gallery__gallery"><div class="tiled-gallery__row"><div class="tiled-gallery__col"><figure class="tiled-gallery__item"><img alt="" data-height="1253" data-id="159" data-link="http://35a97fbe.eu.ngrok.io/?attachment_id=159" data-url="http://35a97fbe.eu.ngrok.io/wp-content/uploads/2019/05/pexels-photo-892548-1024x682.jpeg" data-width="1880" src="https://i2.wp.com/35a97fbe.eu.ngrok.io/wp-content/uploads/2019/05/pexels-photo-892548-1024x682.jpeg"/></figure></div><div class="tiled-gallery__col"><figure class="tiled-gallery__item"><img alt="" data-height="1259" data-id="157" data-link="http://35a97fbe.eu.ngrok.io/?attachment_id=157" data-url="http://35a97fbe.eu.ngrok.io/wp-content/uploads/2019/05/loro-park-orca-fish-1024x686.jpg" data-width="1880" src="https://i0.wp.com/35a97fbe.eu.ngrok.io/wp-content/uploads/2019/05/loro-park-orca-fish-1024x686.jpg"/></figure></div></div></div></div>
<!-- /wp:jetpack/tiled-gallery -->
```

The difference is that images now have `data-height`, `data-width`, `data-link` attributes that `srcset` mechanism relies on:

https://github.com/Automattic/jetpack/blob/9e99c0b665519f83ea85c0200513b0a2a283f09d/extensions/blocks/tiled-gallery/tiled-gallery.php#L60-L61

Curiously, layout is applied image by image so if you have two images, you actually have a state in between where one of them has these attributes and other one doesn't. You can observe this behaviour by switching to "code view" and using undo/redo buttons:

<img width="335" alt="Screenshot 2019-05-10 at 17 51 12" src="https://user-images.githubusercontent.com/87168/57536377-bef5cf80-734c-11e9-963d-18ab5f73b134.png">

**Comment by simison** [comment]
TL;DR: so as long as users interact with the editor after they've transformed the images, all is fine, making the impact for this bug kinda low. :-)

**Comment by simison** [comment]
This might be related — layout doesn't get applied right after style switch. I gets applied when I resize the window little bit:

Video: https://cloudup.com/cPnoKxZPt1P

**Comment by jeherve** [comment]
This was done in #32598.
